mempool/src/core_mempool/index.rs

```diff
@@ -256,33 +257,55 @@ impl TimelineIndex {
 /// (because their sequence number is too high)
 /// we keep separate index to be able to efficiently evict them when Mempool is full
 pub struct ParkingLotIndex {
-    data: BTreeSet<TxnPointer>,
+    // (k, v) = (account, seq nums of txns for this account)
+    data: HashMap<AccountAddress, BTreeSet<u64>>,
+    size: usize,
 }
 
 impl ParkingLotIndex {
     pub(crate) fn new() -> Self {
         Self {
-            data: BTreeSet::new(),
+            data: HashMap::new(),
+            size: 0,
         }
     }
 
     /// add transaction to index
     pub(crate) fn insert(&mut self, txn: &MempoolTransaction) {
-        self.data.insert(TxnPointer::from(txn));
+        if self
+            .data
+            .entry(txn.txn.sender())
+            .or_insert_with(BTreeSet::new)
+            .insert(txn.txn.sequence_number())
+        {
+            self.size += 1;
+        }
     }
 
     /// remove transaction from index
     pub(crate) fn remove(&mut self, txn: &MempoolTransaction) {
-        self.data.remove(&TxnPointer::from(txn));
+        let sender = &txn.txn.sender();
+        if let Some(seq_numbers) = self.data.get_mut(sender) {
+            if seq_numbers.remove(&txn.txn.sequence_number()) {
+                self.size -= 1;
+            }
+            if seq_numbers.is_empty() {
+                self.data.remove(sender);
+            }
+        }
     }
 
     /// returns random "non-ready" transaction (with highest sequence number for that account)
     pub(crate) fn pop(&mut self) -> Option<TxnPointer> {
-        self.data.iter().rev().next().cloned()
+        let mut rng = rand::thread_rng();
+        self.data
+            .iter()
+            .choose(&mut rng)
```

Comment:
Hmmm, from what I can tell this is O(n) complexity.  And if we have a lot of addresses w/ values in the hashmap, I wonder if we should instead use something like https://crates.io/crates/indexmap

Thoughts?